components/org.wso2.identity.event.http.publisher/src/main/java/org/wso2/identity/event/http/publisher/internal/service/impl/HTTPEventPublisherImpl.java

```diff
@@ -0,0 +1,146 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.identity.event.http.publisher.internal.service.impl;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.identity.base.IdentityRuntimeException;
+import org.wso2.carbon.identity.event.publisher.api.exception.EventPublisherException;
+import org.wso2.carbon.identity.event.publisher.api.exception.EventPublisherServerException;
+import org.wso2.carbon.identity.event.publisher.api.model.EventContext;
+import org.wso2.carbon.identity.event.publisher.api.model.SecurityEventTokenPayload;
+import org.wso2.carbon.identity.event.publisher.api.service.EventPublisher;
+import org.wso2.carbon.identity.webhook.management.api.exception.WebhookMgtException;
+import org.wso2.carbon.identity.webhook.management.api.model.Webhook;
+import org.wso2.identity.event.http.publisher.constant.HTTPAdapterConstants;
+import org.wso2.identity.event.http.publisher.api.exception.HTTPAdapterException;
+import org.wso2.identity.event.http.publisher.internal.component.ClientManager;
+import org.wso2.identity.event.http.publisher.internal.component.HTTPAdapterDataHolder;
+import org.wso2.identity.event.http.publisher.internal.util.HTTPCorrelationLogUtils;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import static org.wso2.identity.event.http.publisher.constant.ErrorMessage.ERROR_ACTIVE_WEBHOOKS_NOT_FOUND;
+import static org.wso2.identity.event.http.publisher.constant.ErrorMessage.ERROR_PUBLISHING_EVENT;
+import static org.wso2.identity.event.http.publisher.internal.util.HTTPAdapterUtil.logDiagnosticFailure;
+import static org.wso2.identity.event.http.publisher.internal.util.HTTPAdapterUtil.logDiagnosticSuccess;
+import static org.wso2.identity.event.http.publisher.internal.util.HTTPAdapterUtil.logPublishingEvent;
+import static org.wso2.identity.event.http.publisher.internal.util.HTTPCorrelationLogUtils.handleResponseCorrelationLog;
+
+/**
+ * OSGi service for publishing events using http adapter.
+ */
+public class HTTPEventPublisherImpl implements EventPublisher {
+
+    private static final Log log = LogFactory.getLog(HTTPEventPublisherImpl.class);
+    private List<Webhook> activeWebhooks;
+
+    @Override
+    public String getAssociatedAdapter() {
+
+        return HTTPAdapterConstants.HTTP_ADAPTER_NAME;
+    }
+
+    @Override
+    public void publish(SecurityEventTokenPayload eventPayload, EventContext eventContext)
+            throws EventPublisherException {
+
+        try {
+            makeAsyncAPICall(eventPayload, eventContext);
+            log.debug("Event published successfully to the HTTP.");
```

Comment:
Should indicate the endpoint at least

Reply:
fixed with https://github.com/wso2-extensions/identity-event-publishers/pull/37/commits/15538dc2b9ce588c47e8db9c5db39156de457b3c